### ISSUE TITLE
Adding 'flutter config' for mobile

### ISSUE THREAD

**Issue by csells** [flutter/flutter#58999]
Today, we have configuration options for pre-stable platform support:

```sh
$ flutter config --[no-]enable-web
$ flutter config --[no-]enable-macos-desktop
$ flutter config --[no-]enable-windows-desktop
$ flutter config --[no-]enable-linux-desktop
```

The flutter doctor command uses this information to provide per-platform diagnostics information. However, we don’t have the mobile equivalents:

```sh
$ flutter config --[no-]enable-android
$ flutter config --[no-]enable-ios
```

The consequence of this is that flutter doctor has no way to know whether a Flutter developer is ever going to use their machine for mobile development, so assumes that they are and provides all kinds of scary output directing the user to set up tools and libraries that they may never actually need.

By adding the mobile config options, we enable flutter doctor to tailor it’s output so as to avoid asking developers to do setup work that they do not need to do.

**Comment by jonahwilliams** [comment]
Tentatively assigning for September, also two related issues this would close are, assuming with did this for every platform.

https://github.com/flutter/flutter/issues/12768
https://github.com/flutter/flutter/issues/52859

**Comment by jonahwilliams** [comment]
and: https://github.com/flutter/flutter/issues/44089

Provided that we allowed config to be set before downloading platform dependencies (which it should).

**Comment by jonahwilliams** [comment]
unassigning for now since I'm not actively working on this

**Comment by timsneath** [comment]
Thrilled to see this -- thanks for checking in the PR, @jonahwilliams!

I'm not seeing quite the behavior I expect, though. On a Windows machine, I've disabled Android support, but I still get a warning that Android Studio isn't installed, which I think is inconsistent:

```
[C:\Windows\System32] flutter config --no-enable-android
Setting "enable-android" value to "false".

You may need to restart any open editors for them to read new settings.

[C:\Windows\System32] flutter upgrade
Flutter is already up to date on channel master
Flutter 1.21.0-6.0.pre.214 • channel master • git@github.com:flutter/flutter.git
Framework • revision 6338181428 (38 minutes ago) • 2020-08-05 15:26:10 -0700
Engine • revision e2a6b55e8e
Tools • Dart 2.10.0 (build 2.10.0-3.0.dev d5ff386abe)

[C:\Windows\System32] flutter doctor
Doctor summary (to see all details, run flutter doctor -v):
[✓] Flutter (Channel master, 1.21.0-6.0.pre.214, on Microsoft Windows [Version 10.0.19041.423], locale en-US)
[✓] Chrome - develop for the web
[✓] Visual Studio - develop for Windows (Visual Studio Community 2019 16.6.5)
[!] Android Studio (not installed)
[✓] VS Code (version 1.47.3)
[✓] Connected device (4 available)

! Doctor found issues in 1 category.
```

I _think_ we should be satisfied with the installation I have, right, rather than complaining at the absence of Android Studio? All the prerequisites are satisfied for me to develop web or Windows, which are the only two platforms my install is configured to support. 

The current situation would be analogous to Flutter complaining that I don't have Visual Studio installed when I have the Windows target disabled (which it doesn't do). 

Thoughts, @csells?

**Comment by csells** [comment]
I think the core issue is that both AS and VSC can be used to target any platform that Flutter supports (and that's supported by the host OS), be it web, Windows, iOS, etc. We could decide that if you have one or the other of either AS or VSC that we shouldn't complain about not having the other, since you have a full toolset as far as we're concerned.

**Comment by jonahwilliams** [comment]
I actually think we should remove the IDE validators altogether. There are plenty of other editors that support flutter and dart (even if they aren't 1st party), and we're sort of conflating a "prefered setup" with the other doctor categories that are really "minimum viable" 

Plus we're always going to have problems detecting AS/intellij/Vscode that are installed in unique ways, or updates which change the plugin structure.

**Comment by csells** [comment]
I understand that reliably checking for AS or VSC is an issue but I don't agree that we provide a reasonable Flutter developer experience without one of AS or VSC and the associated Flutter extension.

**Comment by timsneath** [comment]
It's a gray area, not least because I can imagine a world fairly soon where
someone can usefully use Flutter without needing either of our recommended
IDEs. For example, several tools partners we're working with might reach
the point in the next 12 months where they provide all the tooling for an
animator or designer to create a complete Flutter app without needing a
developer tool on their box.

Yet I still lean towards Chris' instinct of enabling some tools validation
-- since for the time being there's a negligible % of our audience who will
be able to achieve their goals without *one* of IntelliJ/AS or VSC, and
plenty of developers who rely on flutter doctor to tell them that their
machine is "ready". But is the algorithm "show a checkmark for *one or
more *IDEs that match the core Flutter profile, but do not show an issue if
an IDE is missing?" What if Android Studio is installed but missing the
Flutter plug-in? Or should we have a separate command for flutter doctor
--tools that specifically tests all IDEs it can find?

Bottom line -- I do very much want to get away from a world where we
presume that every Flutter developer is building (or cares about) a mobile
app. It should be possible for an Ubuntu, Windows or web developer to
install and use Flutter without mobile considerations ever cropping up.

**Comment by jonahwilliams** [comment]
I've recently updated the IDE validation so that it is always non-fatal, this might address some of the concerns here.
